### PR TITLE
fix(ingest/bigquery): fix handling of time decorator offset queries

### PR DESCRIPTION
**Pull request by mayurinehate**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/tests/unit/test_bigquery_source.py [hunk 1]
```diff
@@ -602,6 +602,12 @@ def test_get_table_and_shard_custom_shard_pattern(
         ("project.dataset.table", "project.dataset.table"),
         ("project.dataset.table_20231215", "project.dataset.table"),
         ("project.dataset.table@1624046611000", "project.dataset.table"),
+        ("project.dataset.table@-9600", "project.dataset.table"),
+        ("project.dataset.table@-3600000", "project.dataset.table"),
+        ("project.dataset.table@-3600000--1800000", "project.dataset.table"),
```
Comment (asikowitz) [inline metadata-ingestion/tests/unit/test_bigquery_source.py:607]: What does this double dash even mean lol, just curious

**Review by asikowitz — APPROVED** [review]
LGTM, one suggested change for simplicity, if possible

File: metadata-ingestion/tests/unit/test_bigquery_source.py (hunk 1 above)
Comment (mayurinehate) [reply metadata-ingestion/tests/unit/test_bigquery_source.py:607]: that query is @(time1-time2) where time is represented as negative delta from current time

https://cloud.google.com/bigquery/docs/table-decorators#range-examples

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_audit.py [hunk 2]
```diff
@@ -182,8 +182,9 @@ def __str__(self) -> str:
 class BigQueryTableRef:
     # Handle table time travel. See https://cloud.google.com/bigquery/docs/time-travel
     # See https://cloud.google.com/bigquery/docs/table-decorators#time_decorators
-    # Handling for @0 and @-TIME_OFFSET is apparently missing
-    SNAPSHOT_TABLE_REGEX: ClassVar[Pattern[str]] = re.compile(r"^(.+)@(\d{13})$")
+    SNAPSHOT_TABLE_REGEX: ClassVar[Pattern[str]] = re.compile(
+        "^(.+)@(-?\\d{1,13})(-?(-\\d{1,13})?)?$"
```
[the remark this answers is not in the document]
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_audit.py]: I've updated the regex. Its in the form:
time1, time2 : of regex TIME (-?\\d{1,13})

We should support.
TIME
TIME-TIME
TIME-

The parenthesis should make more sense now.

**Pull request merged**